### PR TITLE
feat: support React 16

### PR DESCRIPTION
**Pull request by neoziro**
- Fix all tests
- Update peerDependencies
- Fix all warnings

### CODE REVIEW

File: src/__tests__/withState.test.js [hunk 1]
```diff
@@ -4,28 +4,34 @@ import { Dummy } from './utils'
 import { compose, withProps, withState } from '../'
 
 describe('withState', () => {
-  it('adds a stateful value and a function for updating it', () => {
+  it.only('adds a stateful value and a function for updating it', () => {
```
Comment (matthieuprat) [inline src/__tests__/withState.test.js]: 🙈

File: src/utils/createHOCFromMapper.js [hunk 2]
```diff
@@ -35,11 +35,7 @@ const createComponentFromMappers = (mappers, childFactory) => {
         .toESObservable(childProps$)
         .subscribe({
           next: childProps => {
-            if (!this.state && !this.mounted) {
-              this.state = { childProps }
-            } else {
-              this.setState({ childProps })
-            }
+            this.setState({ childProps })
```
Comment (matthieuprat) [inline src/utils/createHOCFromMapper.js]: We can also get rid of `componentDidMount()`.

**Comment by matthieuprat** [comment]
👍 

Any chance we can get rid of those `wrapper.update()` that sprang out the code base in the future?

**Comment by neoziro** [comment]
@matthieuprat no it is the new API of enzyme, the tree resolution is not dynamic.

**Comment by matthieuprat** [comment]
#89

**Pull request merged**